### PR TITLE
DOC: Compare 'tolist' function to 'list' in example

### LINKED ISSUE #14683: Doc improvement suggestions: list () and tolist () are very different.
numpy 1.15.4
python 3.7.1

I refer to this doc but I spent a lot of time discovering the following.

https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.tolist.html?highlight=tolist#numpy.ndarray.tolist

```python
import numpy as np

npA=np.uint32([1,2,3])

listA=list(npA)
print(type(listA)) # <class 'list'>
print(type(listA[0])) # <class 'numpy.uint32'>

listA2 = npA.tolist()
print(type(listA2)) # <class 'list'>
print(type(listA2[0])) # <class 'int'>
```

I wanted to calculate numbers greater than 32bit integers, but I should have used tolist() only.

### PR DESCRIPTION
**Pull request by mproszewska**
This closes #14683.

### CODE REVIEW

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -3953,13 +3953,23 @@
...
+    <class 'numpy.uint32'>
+    >>> a_tolist = a.tolist()
+    >>> a_tolist
     [1, 2]
+    >>> type(a_tolist)
+    <class 'list'> 
+    >>> type(a_tolist[0])
+    <class 'int'>
 
     However, for a 2D array, ``tolist`` applies recursively:
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py]:
```suggestion
    Additionally, for a 2D array, ``tolist`` applies recursively:
```

File: numpy/core/_add_newdocs.py [hunk 2]
```diff
@@ -3953,13 +3953,23 @@
...
 
-    >>> a = np.array([1, 2])
-    >>> list(a)
+    >>> a = np.uint32([1, 2])
+    >>> a_list = list(a)
+    >>> a_list
     [1, 2]
-    >>> a.tolist()
+    >>> type(a_list)
+    <class 'list'>
```
Comment (eric-wieser) [inline numpy/core/_add_newdocs.py]: I'd probably skip this one and the one below, it's not particularly interesting

Comment (eric-wieser) [reply numpy/core/_add_newdocs.py]: Fine to leave if no one else agrees with me

File: numpy/core/_add_newdocs.py [hunk 3]
```diff
@@ -3953,13 +3953,23 @@
 
     Examples
     --------
-    For a 1D array, ``a.tolist()`` is almost the same as ``list(a)``:
+    For a 1D array, ``a.tolist()`` is almost the same as ``list(a)``, except that it changes numpy scalars to Python scalars:
```
Comment (mattip) [inline numpy/core/_add_newdocs.py]: line length should be limited to 80 characters. Also, maybe better as "except that ``tolist`` changes ...

File: numpy/core/_add_newdocs.py (hunk 2 above)
Comment (mproszewska) [reply numpy/core/_add_newdocs.py]: I agree and will change it

**Comment by mattip** [comment]
Thanks @mproszewska

**Pull request merged**